File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/library/items/media/files/cached/persistence/IDiskFileAccessTimeUpdater.java
package com.lasthopesoftware.bluewater.client.browsing.library.items.media.files.cached.persistence;

import com.lasthopesoftware.bluewater.client.browsing.library.items.media.files.cached.repository.CachedFile;
import com.namehillsoftware.handoff.promises.Promise;

public interface IDiskFileAccessTimeUpdater {
	Promise<CachedFile> promiseFileAccessedUpdate(CachedFile cachedFile);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/files/cached/persistence/IDiskFileAccessTimeUpdater.kt
package com.lasthopesoftware.bluewater.client.browsing.files.cached.persistence

import com.lasthopesoftware.bluewater.client.browsing.files.cached.repository.CachedFile
import com.namehillsoftware.handoff.promises.Promise

interface IDiskFileAccessTimeUpdater {
    fun promiseFileAccessedUpdate(cachedFile: CachedFile): Promise<CachedFile>
}
